orchagent/fgnhgorch.cpp

```diff
@@ -868,6 +933,11 @@ bool FgNhgOrch::setNewNhgMembers(FGNextHopGroupEntry &syncd_fg_route_entry, FgNh
         }
     }
 
+    if (is_warm_reboot)
```

Comment:
>is_warm_reboot [](start = 8, length = 14)

If `nexthopsMap != m_recoveryMap.end()`, then you will not erase. Is it a bug?

Reply:
if nexthopsMap != m_recoveryMap.end(), it a warm restart case. after recover from the map, we will erase the entry from recovery. nexthopsMap == m_recoveryMap.end() meaning this route is a new entry, will handle in a normal way.